vimPlugins: add jedi-vim

schmittlauch opened the pull request:
This is the master branch equivalent of  #49589, as the vimPlugins structure differs between master and release right now.

running the `updated.py` as described in the default.nix also updated
other vim plugins. If that's not wanted, I can reduce this changeset to
the jedi-vim related changes only.

As this is a python3 plugin it'd be nice to check for python3 support in
the used vim, but apparently nobody else does this.

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

schmittlauch commented:
ping @jagajaga

andir reviewed (changes requested):
What I'd like to see here is that the bump of the other plugins in a commit separated from the addition of jedi-vim.

schmittlauch commented:
But the version bumps can still be included, just separated to another commit?

andir commented:
Yes, They will either happen now or someone will do them later... I do not see a test framework to verify that they actually work..

schmittlauch committed:
vim-plugins: add jedi-vim

As this is a python3 plugin it'd be nice to check for python3 support in
the used vim, but apparently nobody else does this.

schmittlauch committed:
vim-plugins: automatic version bumps

The recommended ./update.py script for adding new plugins also bumps the
version of already existing plugins.

schmittlauch commented:
@andir splitting is done.

andir reviewed (changes requested):
Looks good just some nitpicks.

schmittlauch commented on pkgs/misc/vim-plugins/overrides.nix:
```diff
@@ -334,4 +334,15 @@ with generated;
+    meta = {
+      description = "code-completion for python using python-jedi";
+      license = stdenv.lib.licenses.mit;
+  };
```
fixed

schmittlauch commented on pkgs/misc/vim-plugins/overrides.nix:
```diff
@@ -334,4 +334,15 @@ with generated;
+      license = stdenv.lib.licenses.mit;
+  };
+  });
+
```
fixed
